Package.swift

```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.4
+// swift-tools-version:5.6
```

Comment:
Strangely, this change now __requires__ that `--enable-test-discovery` be passed when compiling the tests on linux/Android, or [it errors like this with 5.6, 5.7, and trunk](https://github.com/buttaface/swift-android-sdk/runs/7742640026?check_suite_focus=true#step:13:496):
```
swift-nio/Tests/LinuxMain.swift:146:1: error: expressions are not allowed at the top level
(LinuxMainRunnerImpl() as LinuxMainRunner).run()
```
Anyone know what's going on?

Reply:
I believe this is a bug in SwiftPM.

Reply:
Alright, I'll file a bug for SPM.

Reply:
@Lukasa Now that NIO supports 5.4 as a minimum can LinuxMain.swift not just be removed?

Reply:
Sadly no. NIO tests with `--warnings-as-errors`, and we have number of tests for deprecated code. The test discovery stubs in SwiftPM generate warnings when those tests are present up until 5.6, which is why our CI continues to use `LinuxMain.swift` for 5.4 and 5.5. But we are at least _finally_ on a path to dropping it.

Reply:
Filed, apple/swift-package-manager#5717.

Reply:
this _should_ be fixed in 5.7 and main. but perhaps not available in the nightly toolchain yet?

Reply:
It is fixed with the latest Aug. 9 trunk snapshot build for linux, but I'm using a June trunk snapshot because of another breaking regression for me. As you noted in that issue, should only be a problem with 5.6 now.

Reply:
yea will be hard to justify pulling into 5.6 given workaround.

Reply:
Which workaround? Adding @main is a non-starter if someone is testing 5.3 and earlier, when the attribute wasn't there, apple/swift@df99de8. As for overriding it with `--enable-test-discovery`, adding that only for 5.6.3 for all future CI testing is going to be annoying and has other consequences.